Classes/Settings.cs

```diff
@@ -490,5 +481,32 @@ protected static void SetValue(string propertyName, string value)
                 }
             }
         }
+
+        private static string GetValuePostfix(Model.Programme progInfo, string name, string defaultValue)
+        {
+            string value = null;
+            if (progInfo != null)
+            {
+                value = GetValue(name + progInfo.Progid.ToString(CultureInfo.InvariantCulture));
```

Comment:
> delete from settings where progid not null and property="SaveFolder"

This is the main reason for separating it out IMO.

I don't have an issue with null fields.  However, managing the per-programme settings via the Programme class and perhaps storing the values in a separate table seems the way to go for me.